django/applications/catmaid/static/js/label-annotations.js

```diff
@@ -0,0 +1,171 @@
+(function (CATMAID) {
+
+  "use strict";
+
+  class LabelAnnotations {
+    constructor() {
+      this.groupManagingStack = new Map();
+      this.managers = new Map();
+      this.active = undefined;
+    }
+
+    activate(stackID) {
+      this.active = stackID;
+    }
+
+    clear() {
+      for (const [_stack, manager] of this.managers.entries()) manager.unregister();
+      this.managers.clear();
+      this.groupManagingStack.clear();
+      this.active = undefined;
+    }
+
+    get(stack) {
+      let manager = this.managers.get(stack.id);
+
+      if (!manager) {
+        // Find all stack groups to which this stack belongs.
+        return CATMAID.fetch(project.id + '/stack/' + stack.id + '/groups')
+            .then(response =>
+              // Get the info of all of these groups.
+              Promise.all(response.stack_group_ids.map(
+                  sg_id => CATMAID.fetch(project.id + '/stackgroup/' + sg_id + '/info')))
+            )
+            .then(sg_infos => {
+              // Find stack groups where this stack is an (ortho)view.
+              let ortho_sgs = sg_infos
+                  .filter(sg_info => sg_info.stacks
+                      .find(s => s.id === stack.id)
+                      .relation === 'view');
+
+              // Find if an existing manager for any of these stack groups exists.
+              let managing_stack_id = ortho_sgs.find(sg => this.groupManagingStack.get(sg.id));
+
+              if (typeof managing_stack_id !== 'undefined') {
```

Comment:
Out of curiosity: Is there any benefit in using this over `managing_stack_id !== undefined` (which seems easier to read to me)?

Reply:
Since `managing_stack_id` is declared in this scope, no benefit. Will fix.